plugin-server/src/worker/ingestion/process-event.ts

```diff
@@ -286,6 +287,58 @@ export interface SummarizedSessionRecordingEvent {
     message_count: number
 }
 
+type ConsoleLogEntry = {
+    team_id: number
+    message: string
+    log_level: 'info' | 'warn' | 'error'
+    log_source: 'session_replay'
+    // the session_id
+    log_source_id: string
+    // TODO will we ever emit this?
+    instance_id: string
+    timestamp: ClickHouseTimestamp
+}
+
+function safeString(payload: string[]) {
+    // the individual strings are sometimes wrapped in quotes... we want to strip those
+    return payload
+        .map((item) => {
+            let candidate = item
+            if (candidate.startsWith('"') || candidate.startsWith("'")) {
+                candidate = candidate.substring(1)
+            }
+
+            if (candidate.endsWith('"') || candidate.endsWith("'")) {
+                candidate = candidate.substring(0, candidate.length - 1)
+            }
+            return candidate
+        })
+        .join(' ')
+}
+
+export const gatherConsoleLogEvents = (team_id: number, session_id: string, events: RRWebEvent[]) => {
+    const consoleLogEntries: ConsoleLogEntry[] = []
+
+    events.forEach((event) => {
+        if (event.type === 6 && event.data?.plugin === 'rrweb/console@1') {
```

Comment:
What does the 6 relate to here? Might be worth extracting to a variable

Reply:
Ah, yep, some rrweb magic - we might already have this as a constant somewhere close by tbh